ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsTask.java

```diff
@@ -118,11 +119,17 @@ public String getName() {
     private boolean isMissingAcidState = false;
     private BasicStatsWork work;
     private boolean followedColStats1;
+    private Map<String, String> providedBasicStats;
 
     public BasicStatsProcessor(Partish partish, BasicStatsWork work, HiveConf conf, boolean followedColStats2) {
       this.partish = partish;
       this.work = work;
       followedColStats1 = followedColStats2;
+      Table table = partish.getTable();
+      if (table.isNonNative()) {
+        TableDesc tableDesc = Utilities.getTableDesc(table);
+        providedBasicStats = table.getStorageHandler().getBasicStatistics(tableDesc);
+      }
```

Comment:
this `Processor` is created for a `Partish` and not for a table - this part will work incorrectly for partitions.